.github/workflows/release.yml

```diff
@@ -0,0 +1,55 @@
+name: Automated product release
+
+on:
+  pull_request:
+    branches: [master]
+    types: [closed]
+
+jobs:
+  check-release:
+    name: Check release required
+    runs-on: ubuntu-latest
+    if: github.event.pull_request.merged && endsWith(github.repository, '-private') != true
+    outputs:
+      release: ${{ steps.check.outputs.ready }}
+    steps:
+      - name: Checkout actions
+        uses: actions/checkout@v3
+        with:
+          repository: pubnub/client-engineering-deployment-tools
+          ref: v1
+          token: ${{ secrets.GH_TOKEN }}
+          path: .github/.release/actions
+      - id: check
+        name: Check pre-release completed
+        uses: ./.github/.release/actions/actions/checks/release
+        with:
+          token: ${{ secrets.GH_TOKEN }}
+  publish:
+    name: Publish package
+    runs-on: ubuntu-latest
+    needs: check-release
+    if: needs.check-release.outputs.release == 'true'
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+        with:
+          # This should be the same as the one specified for on.pull_request.branches
+          ref: master
+      - name: Checkout actions
+        uses: actions/checkout@v3
+        with:
+          repository: pubnub/client-engineering-deployment-tools
+          ref: v1
+          token: ${{ secrets.GH_TOKEN }}
+          path: .github/.release/actions
+      - name: Prepare cargo environment
+        shell: bash
+        run: cargo install cargo-readme
+      - name: Publish to Crates
+        env:
+          CARGO_REGISTRY_TOKEN: ${{ secrets.CARGO_REGISTRY_TOKEN }}
```

Comment:
It will work without cargo login?
Or this actions contain `cargo login` command embedded into it?

Reply:
From docs: 
> --token token
API token to use when authenticating. This overrides the token stored in the credentials file (which is created by [cargo-login(1)](https://doc.rust-lang.org/cargo/commands/cargo-login.html)).
[Cargo config](https://doc.rust-lang.org/cargo/reference/config.html) environment variables can be used to override the tokens stored in the credentials file. The token for crates.io may be specified with the CARGO_REGISTRY_TOKEN environment variable. Tokens for other registries may be specified with environment variables of the form CARGO_REGISTRIES_NAME_TOKEN where NAME is the name of the registry in all capital letters.